### PR TITLE
[BEAM-1984] Fix scope for dependencies needed only for test/runtime purposes

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by iemejia** [comment]
R: @aviemzur @dhalperi 
CC: @tweise for the malhar dep (in case of doubt).

File: sdks/java/io/elasticsearch/pom.xml [hunk 1]
```diff
@@ -86,18 +87,19 @@
...
     <dependency>
       <groupId>com.google.auto.value</groupId>
       <artifactId>auto-value</artifactId>
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
+      <scope>provided</scope>
```
Comment (dhalperi) [inline sdks/java/io/elasticsearch/pom.xml]: Why provided? I do not believe we necessarily require runners to provide the `joda-time` jar.

File: sdks/java/io/jdbc/pom.xml [hunk 2]
```diff
@@ -89,6 +89,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
+      <scope>test</scope>
```
Comment (dhalperi) [inline sdks/java/io/jdbc/pom.xml]: Since this is not used at compile time, why is the build passing?

File: sdks/java/io/jdbc/pom.xml [hunk 3]
```diff
@@ -107,18 +108,19 @@
...
     <dependency>
       <groupId>com.google.auto.value</groupId>
       <artifactId>auto-value</artifactId>
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
+      <scope>provided</scope>
```
Comment (dhalperi) [inline sdks/java/io/jdbc/pom.xml]: this is just `test`, isn't it?

File: sdks/java/io/jdbc/pom.xml (hunk 2 above)
Comment (iemejia) [inline sdks/java/io/jdbc/pom.xml]: The dependency was defined for the compile scope, but it was only used by the test classes so we can refine the scope to specific to test.

Comment (dhalperi) [inline sdks/java/io/jdbc/pom.xml]: Yes, but shouldn't that register as an `unused declared dependency` and fail the build?

File: sdks/java/extensions/gcp-core/pom.xml [hunk 4]
```diff
@@ -150,6 +151,12 @@
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <scope>provided</scope>
```
Comment (aviemzur) [inline sdks/java/extensions/gcp-core/pom.xml]: Can we add a comment here on why this is provided and who provides it?

File: sdks/java/io/google-cloud-platform/pom.xml [hunk 5]
```diff
@@ -139,6 +139,7 @@
     <dependency>
       <groupId>io.grpc</groupId>
       <artifactId>grpc-stub</artifactId>
+      <scope>provided</scope>
```
Comment (aviemzur) [inline sdks/java/io/google-cloud-platform/pom.xml]: Can we add a comment here on why this is provided and who provides it?

**Comment by dhalperi** [comment]
Hey @iemejia I know you're doing more work here -- any updates?

**Comment by iemejia** [comment]
Yes, it is a pity we didn't discuss this particular PR at Apachecon, I will redo the PR once I can but remember this is part 1 of 2, the second part is where I expect some debate. I will ping you back once I check this again.

**Comment by aaltay** [comment]
Hey @iemejia, are you going to keep this PR or open a new one ?

**Comment by iemejia** [comment]
I will keep this one, haven't had the time to address some of the points but I will.

**Pull request closed without merging**